Cabal/Distribution/Simple/Program/Builtin.hs

```diff
@@ -309,6 +311,16 @@ hscolourProgram = (simpleProgram "hscolour") {
         _         -> ""
   }
 
+doctestProgram :: Program
+doctestProgram = (simpleProgram "doctest") {
+    programFindLocation = \v p -> findProgramOnSearchPath v p "doctest"
+  , programFindVersion  = findProgramVersion "--version" $ \str ->
+         -- "doctest version 0.11.2"
```

Comment:
Hmm... we kind of really want the GHC version too, so you can check for compatibility. Not sure how to fix the API to make this possible.